### PR TITLE
feat: allow hljs to be run from command line

### PR DESCRIPTION
**Pull request by marcoscaceres**
This allows the ability for hljs's build script to be run from
the command line.

This is useful for sofware that requires hljs as part of a build
chain.

It disables generation of docs and demo by default.

The patch adds the following options to "tools/build.js":

 * `--output`: path to write build files to. Defaults to "build".
 * `--docs`: tells builder to include the docs and demo.

Build just for Web, don't include docs:

```Bash
hljs --output "some/path xml" javascript
```

Build for Web, include docs:

```Bash
hljs --docs --output "some/path" xml javascript
```

Build for NodeJS:

```Bash
hljs -t node --output "some/path" xml javascript
```

Build for everything:

```Bash
hljs -t all --output "some/path" xml javascript
```

### CODE REVIEW

**Comment by marcoscaceres** [comment]
Good point about the docs. I'll add some documentation before merging.

**Comment by marcoscaceres** [comment]
@egor-rogov any idea how we would update https://highlightjs.readthedocs.io/ ?

**Comment by egor-rogov** [comment]
> any idea how we would update https://highlightjs.readthedocs.io/ ?

Hmm, I thought you have access to upload fresh docs, no?

File: package.json [hunk 1]
```diff
@@ -21,6 +21,9 @@
     "url": "git://github.com/highlightjs/highlight.js.git"
   },
   "main": "./lib/index.js",
+  "bin": {
+    "hljs": "./tools/build.js"
```
Comment (fengmk2) [inline package.json]: missing tools/build.js file on npm package. https://github.com/highlightjs/highlight.js/issues/1984

**Pull request merged**